core/activity_pub.md

```diff
@@ -6,35 +6,46 @@ This document specifies the extensions to Fora required for a community to be co
+The DIDs of a client must be recorded on a verifiable data registry, typically this will be a public blockchain or ledger. All communities will rely on client DIDs to resolve the authentication method used by a particular client. Communities may choose not to rely on DIDs for authentication of governance module interaction (NOTE: the authentication method of a client may be different from its corresponding controller, the latter would be used to authenticate a governance vote). In this case, public keys for each client controller must be registered on the community. If the communities chooses to use on-chain DID verification, the client DID controllers must be verifiable via a communication protocol such as InterBlockchain Communication.
 
-Fora implementations which rely on the public key registered with a DID instead of registering a public key with a governance module, will benefit from dynamic swapping of public keys. 
+Client DIDs are intentionally isolated from the communities they are members of in order to allow for individual exit to different communities as well as improved operational security. DIDs allow for clients to rotate private/public key pairs without notifiying all the relevant parties.  The authentication method specified by a client DID must be a form of public/private key authentication.
 
-Meanwhile communities will also have DIDs that identify them to their users. This DID will also be registered on the blockchain as a government. The DID will resolve to the current hosts of the platform. Any governance decisions which affects the set of hosts will update the DID resolving result to output the new set of hosts.
+Communities will also have DIDs that identify them to their users. This DID will also be recorded on a blockchain as a government. The DID Document for this DID will be deterministically generated based on the community parameters that exist at the block height the DID resolving function is being executed.  All governance decisions will be recorded on the DID Document. The DID controller will be the community members and the verification method to update the DID Document will be the passing of a governance proposal. The DID may include a default set of "seed" hosts to allow users to conveniently find and connect to the community which will be set under the service endpoint section of a DID Document. Additional parameters will be added to the DID Document which indicate a list of admittors, enforcers, members, governance decisions and any other relevant information. 
```

Comment:
Does the service endpoint only return the list of hosts? Can I use the DID to query for the admittors, enforcers, community parameters, constitution, etc?

Reply:
Yes, this depends on your DID implementation. [Service endpoints](https://www.w3.org/TR/did-core/#service-endpoints) are natively supported. 

Yes you can, the DID document for the community would include extra [parameters](https://www.w3.org/TR/did-core/#did-parameters) for these keys. Then if you wanted one of those values, you would enter the DID, as a DID URL
`did:cosmoshub5:vov.camp?enforcers=all`

Reply:
anything I need to add to the doc?